packages/datadog-instrumentations/src/aerospike.js

```diff
@@ -0,0 +1,88 @@
+'use strict'
+
+const {
+  channel,
+  addHook
+} = require('./helpers/instrument')
+const { NODE_MAJOR } = require('../../../version')
+const shimmer = require('../../datadog-shimmer')
+
+const startCh = channel('apm:aerospike:command:start')
+const endCh = channel('apm:aerospike:command:end')
+const errorCh = channel('apm:aerospike:command:error')
+
+const asyncStartChannel = channel('apm:aerospike:command:asyncStart')
+const asyncEndChannel = channel('apm:aerospike:command:asyncEnd')
+
+function wrapCreateCommand (createCommand) {
+  if (typeof createCommand !== 'function') return createCommand
+
+  return function commandWithTrace () {
+    const CommandClass = createCommand.apply(this, arguments)
+
+    if (!CommandClass) return CommandClass
+
+    shimmer.wrap(CommandClass.prototype, 'process', wrapProcess)
+
+    return CommandClass
+  }
+}
+
+function wrapProcess (process) {
+  return function (...args) {
+    if (typeof args[0] !== 'function') return process.apply(this, args)
+
+    const ctx = {
+      commandName: this.constructor.name,
+      commandArgs: this.args,
+      clientConfig: this.client.config
+    }
+
+    return startCh.runStores(ctx, () => {
```

Comment:
Why is the long form used here for the channels? Doesn't `TracingChannel` support this use case perfectly?

Reply:
except for the explicit error handling, where you have to check if the callback's first arg is not null which means there was an error and so it needs to be published to the subscriber

Reply:
TracingChannel does that: https://github.com/DataDog/dc-polyfill/blob/56954afcd6c7f2dbabee30d861c89d52e61a1d3d/patch-tracing-channel.js#L136